src/connectors/yandex-music.ts

```diff
@@ -77,167 +106,228 @@ function setupOldConnector(): void {
 	Connector.getCurrentTime = (): number | null => {
 		const el = document.querySelector('.progress__bar.progress__text');
 		const timeStr = el?.getAttribute('data-played-time');
-		return timeStr ? parseFloat(timeStr) : null;
+		if (!timeStr) {
+			return null;
+		}
+		const val = parseFloat(timeStr);
+		return isNaN(val) ? null : val;
 	};
 
 	Connector.getDuration = (): number | null => {
 		const el = document.querySelector('.progress__bar.progress__text');
 		const durStr = el?.getAttribute('data-duration');
-		return durStr ? parseFloat(durStr) : null;
+		if (!durStr) {
+			return null;
+		}
+		const val = parseFloat(durStr);
+		return isNaN(val) ? null : val;
 	};
 
 	Connector.isPlaying = (): boolean => {
-		const btn = document.querySelector('.player-controls__btn_play');
-		return btn?.classList.contains('player-controls__btn_pause') ?? false;
+		const btn2 = document.querySelector('.player-controls__btn_play');
+		return btn2?.classList.contains('player-controls__btn_pause') ?? false;
 	};
 }
 
-// NEW CONNECTOR
-
-function setupNewConnector(): void {
-	Connector.playerSelector =
-		"section[class*='PlayerBarDesktopWithBackgroundProgressBar_root']";
-
-	Connector.getTrack = (): string | null => {
+function setupNewConnector() {
+	Util.debugLog(`${connectorLabel}: Initialization started`, 'log');
+	let attempts = 0;
+	const maxAttempts = 20;
+	const intervalMs = 500;
+	const initInterval = setInterval(() => {
```

Comment:
What's the reason for using `setInterval` vs adding a `MutationObserver` to check if the DOM contains the correct elements?